backend/core/src/csv/csv-builder.service.ts

```diff
@@ -1,130 +1,330 @@
+        "K"
+      ),
+      income: "_I_Income",
+      accessibility: "_L_ADA",
+      incomeVouchers: "_M_Vouchers or Subsidies",
+      preferredUnit: "_N_Requested Unit Type",
+      preferences: "_O_Preferences",
+      householdSize: "_P_Household Size",
+      householdMembers: "_Q_Household Members",
+      markedAsDuplicate: "_R_Marked As Duplicate",
+      flagged: "_S_Flagged As Duplicate",
+      demographics: "_T_Demographics",
+    }
+  }
+
+  private mapAddressFields(oldKey: string, newKey: string, sort) {
+    const obj = {}
+    const fields = [
+      "street",
+      "street2",
+      "city",
+      "zipCode",
+      "county",
+      "state",
+      "placeName",
+      "latitude",
+      "longitude",
+    ]
+    fields.forEach((field, i) => {
+      obj[`${oldKey} ${field}`] = `_${sort}${i}_${newKey} ${capitalizeFirstLetter(field)}`
+    })
 
-  private incrementMetadataArrayItemLabel(item: FormattingMetadata, index: number) {
-    const newItem = { ...item }
-    newItem.label += ` (${index})`
-    return newItem
+    return obj
   }
 
-  private retrieveValueByDiscriminator(obj, discriminator: string) {
-    let value: any = obj
-    for (const key of discriminator.split(".")) {
-      if (!key) {
-        continue
-      }
-      value = value[key]
-    }
-    return value
+  private mapPrimaryApplicantFields(sort: string) {
+    const obj = {}
+    const fields = [
+      "firstName",
+      "middleName",
+      "lastName",
+      "birthMonth",
+      "birthDay",
+      "birthYear",
+      "emailAddress",
+      "phoneNumber",
+      "phoneNumberType",
+    ]
+    fields.forEach((field, i) => {
+      obj[`applicant ${field}`] = `_${sort}${i}_Primary Applicant ${this.capAndSplit(field)}`
+    })
+
+    return obj
   }
 
-  private flattenJson(obj: any, metadataAggregate: FormattingMetadataAggregate): Array<string> {
-    let outputRows: string[] = []
-    for (const metadataObj of metadataAggregate) {
-      let value
-      try {
-        value = this.retrieveValueByDiscriminator(obj, metadataObj.discriminator)
-      } catch (e) {
-        value = undefined
+  private mapAlternateContactFields(sort: string) {
+    const obj = {}
+    const fields = [
+      "firstName",
+      "middleName",
+      "lastName",
+      "type",
+      "agency",
+      "otherType",
+      "emailAddress",
+      "phoneNumber",
+    ]
+
+    fields.forEach((field, i) => {
+      obj[`alternateContact ${field}`] = `_${sort}${i}_Alternate Contact ${this.capAndSplit(field)}`
+    })
+
+    return obj
+  }
+
+  private capAndSplit(str: string): string {
+    let newStr = capitalizeFirstLetter(str)
+    newStr = newStr.split(/(?=[A-Z])/).join(" ")
+    return newStr
+  }
+
+  private setHeader(header: string, index: number): [string, string?] {
+    let newKey = header
+    // we pass along the mapped key for reference, so we can more easily set data to the correct column
+    const test1 = this.mappedFields[header]
```

Comment:
Could you write an example in a comment here? It's hard to determine what is going to happen here and what is expected